### PR TITLE
raftstore: adjust leader metircs

### PR DESCRIPTION
**Pull request by Connor1996**
## What have you changed? (mandatory)
After adding threaded raftstore, it is hard to collect the leader count in raftstore. So record this metrics in `region_info_accessor` which has a global view of all the regions of this store.

## What are the type of the changes? (mandatory)
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
manual-test

## Does this PR affect documentation (docs) update? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
https://github.com/pingcap/tidb-ansible/pull/668

### CODE REVIEW

File: src/raftstore/coprocessor/region_info_accessor.rs [hunk 1]
```diff
@@ -439,6 +442,28 @@ impl Runnable<RegionCollectorMsg> for RegionCollector {
     }
 }
 
+const METRICS_FLUSH_INTERVAL: u64 = 10; // 10s
```
Comment (siddontang) [inline src/raftstore/coprocessor/region_info_accessor.rs]: Mostly, we use milliseconds for the interval.

File: src/raftstore/coprocessor/metrics.rs [hunk 2]
```diff
@@ -11,17 +11,25 @@
...
+
     pub static ref REGION_KEYS_HISTOGRAM: Histogram = register_histogram!(
         "tikv_raftstore_region_keys",
         "Bucketed histogram of approximate region keys.",
         exponential_buckets(1.0, 2.0, 30).unwrap()
     ).unwrap();
+
+    pub static ref REGION_COLLECTOR_COUNT_GAUGE_VEC: IntGaugeVec =
+    register_int_gauge_vec!(
+        "tikv_raftstore_region_collector_region_count",
```
[the remark this answers is not in the document]
Comment (zhangjinpeng1987) [reply src/raftstore/coprocessor/metrics.rs]: Agree.

**Comment by Connor1996** [comment]
PTAL

**Pull request merged**